### PR TITLE
feat(start): prewarm vmux://start for instant tab/pane opens

### PR DESCRIPTION
**Pull request by JunichiSugiura**
## Context

Opening `vmux://start/` — the default content of every new tab, pane, stack, and space — showed a visible cold-load delay while a fresh CEF browser was created and the page + WASM loaded. That made the most common "new surface" action feel sluggish.

## Implementation

Keep a small warm pool (default 1) of prewarmed `vmux://start/` webviews parked under a hidden, zero-size node: their CEF browser and WASM load at boot but never composite (a `Visibility::Hidden` ancestor makes them non-renderable, so `sync_children_to_ui` collapses them and CEF hides the native view — near-zero idle cost).

On a start open, `handle_start_page_open` reparents a warm spare into the target stack — instant, since the browser already exists — then refills the pool. It falls back to the existing cold path when the pool is empty, so there is no regression. A parked spare captured boot-time launcher data, so on reveal the host pushes current tabs/spaces/history and refocuses the input, matching a cold open.

Notably required **no changes to `vmux_core` or `vmux_browser`**: reparenting a windowed CEF browser between stacks repositions cleanly via the existing per-frame windowed-sync (teardown only fires on despawn or a windowed↔OSR flip, never on a parent change), and the single-window model means one pool serves every space. Pool size is the `WARM_START_POOL_SIZE` constant.

Design notes in `docs/specs/2026-06-30-warm-start-prewarm-design.md`.

## Checks / QA

- Open new tabs / panes / stacks / spaces — start page should paint near-instantly vs the prior cold delay.
- First open right after launch may still be cold (pool fills at boot) — expected.
- Rapid back-to-back opens (pool = 1) can outrun refill → cold; bump `WARM_START_POOL_SIZE` to 2 if noticeable.
- Pick an entry from a freshly-opened start page → it opens in place and the launcher list reflects current tabs/spaces (not stale boot-time state).

## Summary by CodeRabbit

* **New Features**
  * Added warm-start prewarming to reuse preloaded `vmux://start/` views when available.
  * Maintains a background pool of parked spares and reveals them on demand with refreshed launcher data and keyboard focus.

* **Bug Fixes**
  * Ensures only “ready” warm views are claimed; otherwise falls back to the existing cold-start path.
  * Improves correctness for start-page refresh and focus handling during reuse.

* **Documentation**
  * Added a warm-start prewarm design specification covering behavior and edge cases.

* **Tests**
  * Extended coverage for warm spare reuse, empty-pool fallback, and pool sizing to the configured target.

### CODE REVIEW

File: crates/vmux_layout/src/start/plugin.rs [hunk 1]
```diff
@@ -25,37 +56,130 @@ impl Plugin for StartPlugin {
...
+            commands.entity(task.stack).insert(PageMetadata {
+                url: START_PAGE_URL.to_string(),
+                title: "Start".to_string(),
+                ..default()
+            });
+            commands
+                .entity(spare)
+                .insert((ChildOf(task.stack), CefKeyboardTarget))
+                .remove::<WarmStartSpare>();
+            revealed.write(StartSpareRevealed { webview: spare });
```
[the remark this answers is not in the document]
Comment (JunichiSugiura) [reply crates/vmux_layout/src/start/plugin.rs]: Fixed in f5a900d9. Added a `WarmStartReady` marker, set only when a spare's page actually mounts (it emits `StartDataRequest` on mount). `handle_start_page_open` now claims only spares carrying both `WarmStartSpare` and `WarmStartReady`; an unready spare stays parked and the open falls back to the cold path. Covered by the new `not_ready_spare_is_not_claimed` test.

**Comment by JunichiSugiura** [comment]
@coderabbitai resolve
